components/tools/OmeroWeb/omeroweb/webclient/views.py

```diff
@@ -605,43 +605,52 @@ def api_container_list(request, conn=None, **kwargs):
     # parents), screens and plates (without parents). This is fine for
     # the first page, but the second page may not be what is expected.
 
+    r = dict()
     try:
         # Get the projects
-        projects = tree.marshal_projects(conn=conn,
-                                         group_id=group_id,
-                                         experimenter_id=experimenter_id,
-                                         page=page,
-                                         limit=limit)
+        r['projects'] = tree.marshal_projects(
+            conn=conn,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
 
         # Get the orphaned datasets (without project parents)
-        datasets = tree.marshal_datasets(conn=conn,
-                                         orphaned=True,
-                                         group_id=group_id,
-                                         experimenter_id=experimenter_id,
-                                         page=page,
-                                         limit=limit)
+        r['datasets'] = tree.marshal_datasets(
+            conn=conn,
+            orphaned=True,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
 
         # Get the screens for the current user
-        screens = tree.marshal_screens(conn=conn,
-                                       group_id=group_id,
-                                       experimenter_id=experimenter_id,
-                                       page=page,
-                                       limit=limit)
+        r['screens'] = tree.marshal_screens(
+            conn=conn,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
 
         # Get the orphaned plates (without project parents)
-        plates = tree.marshal_plates(conn=conn,
-                                     orphaned=True,
-                                     group_id=group_id,
-                                     experimenter_id=experimenter_id,
-                                     page=page,
-                                     limit=limit)
-
+        r['plates'] = tree.marshal_plates(
+            conn=conn,
+            orphaned=True,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
         # Get the orphaned images container
-        orphaned = tree.marshal_orphaned(conn=conn,
-                                         group_id=group_id,
-                                         experimenter_id=experimenter_id,
-                                         page=page,
-                                         limit=limit)
+        if request.session['server_settings']['ui']['orphans']:
```

Comment:
Slightly uncomfortable with evaluating an empty dict() as False here.
Since you have 3 settings
```
omero.client.ui.tree.orphans.enabled
omero.client.ui.tree.orphans.name
omero.client.ui.tree.orphans.description
```
they might logically map to a session dict 
```request.session['server_settings']['ui']['orphans']``` with keys of ```enabled```, ```name``` & ```description```.

Then you could (if needed) still get the name & description even if orphans was disabled.